arrows/core/transform_detected_object_set.cxx

```diff
@@ -0,0 +1,275 @@
+  this->dest_camera =\
+    kwiver::vital::read_krtd_file( this->dest_camera_krtd_file_name );
+}
+
+
+// ------------------------------------------------------------------
+bool
+transform_detected_object_set::
+check_configuration( vital::config_block_sptr config ) const
+{
+  kwiver::vital::config_difference cd( this->get_configuration(), config );
+  const auto key_list = cd.extra_keys();
+
+  if ( ! key_list.empty() )
+  {
+    LOG_WARN( logger(), "Additional parameters found in config block that are "
+	      "not required or desired: "
+              << kwiver::vital::join( key_list, ", " ) );
+  }
+
+  return true;
+}
+
+
+// ------------------------------------------------------------------
+vector_3d
+transform_detected_object_set::
+backproject_to_ground( const kwiver::vital::camera_perspective_sptr camera,
+		       const vector_2d img_pt ) const
+{
+  auto ground_plane = vector_4d(0, 0, 1, 0);
+  return this->backproject_to_plane( camera, img_pt, ground_plane );
+}
+
+// ------------------------------------------------------------------
+vector_3d
+transform_detected_object_set::
+backproject_to_plane( const kwiver::vital::camera_perspective_sptr camera,
+		      const vector_2d img_pt,
+		      const vector_4d plane ) const
+{
+  // Back an image point to a specified world plane
+  vector_2d npt_ = camera->intrinsics()->unmap(img_pt);
+  auto npt = vector_3d(npt_(0), npt_(1), 1.0);
+
+  matrix_3x3d M = camera->rotation().matrix().transpose();
+
+  auto n = vector_3d(plane(0), plane(1), plane(2));
+  double d = plane(3);
+
+  vector_3d Mt = M * camera->translation();
+  vector_3d Mp = M * npt;
+
+  return Mp * (n.dot( Mt ) - d) / n.dot( Mp ) - Mt;
+}
+
+// ------------------------------------------------------------------
+Eigen::Matrix<double, 8, 3>
+transform_detected_object_set::
+backproject_bbox( const kwiver::vital::camera_perspective_sptr camera,
+		  const vector_4d box ) const
+{
+  // project base of the box to the ground
+  auto base_pt = vector_2d((box(0) + box(2)) / 2, box(3));
+  vector_3d pc = this->backproject_to_ground( camera, base_pt );
+  vector_3d ray = pc - camera->center();
+  ray(2) = 0.0;
+  ray.normalize();
+
+  vector_3d p1 =\
+    this->backproject_to_ground( camera, vector_2d(box(0), box(3)) );
+  vector_3d p2 =\
+    this->backproject_to_ground( camera, vector_2d(box(2), box(3)) );
+
+  vector_3d vh = p2 - p1;
+  vector_3d vd = ray * vh.norm();
+  vh = vector_3d(-vd(1), vd(0), 0);
+  p1 = pc - vh / 2;
+  p2 = pc + vh / 2;
+  vector_3d p3 = p2 + vd;
+  vector_3d p4 = p1 + vd;
+
+  // TODO: Check that norm of vd is not equal to 0.0?
+
+  vector_3d n = vd.normalized();
+  double d = -(n.transpose() * p3)(0);
+
+  vector_4d back_plane;
+  back_plane << n, d;
+  vector_3d p5 =\
+    this->backproject_to_plane( camera, vector_2d(box(0), box(1)), back_plane );
+  double height = p5(2);
+
+  Eigen::Matrix<double, 8, 3> box3d;
+  box3d << p1.transpose(),
+    p2.transpose(),
```

Comment:
I would indent so that `p2` is below `p1` and use the same indentation for following lines.